src/core/Observer.ts

```diff
@@ -0,0 +1,322 @@
+  /**
+   * Room ID for the realtime subscription on the collection of observed documents
+   */
+  public roomId: string = null;
+
+  /**
+   * Gets documents IDs
+   */
+  ids () {
+    return this.values();
+  }
+
+  /**
+   * Gets Koncorde filters for observed documents
+   */
+  get filters () {
+    return {
+      ids: { values: Array.from(this.values()) }
+    };
+  }
+}
+
+type DocumentUrn = string;
+type CollectionUrn = string;
+
+function documentUrn (index: string, collection: string, id: string): DocumentUrn {
+  return `${index}:${collection}:${id}`;
+}
+
+function collectionUrn (index: string, collection: string): CollectionUrn {
+  return `${index}:${collection}`;
+}
+
+/**
+ * The Observer class allows to manipulate realtime documents.
+ *
+ * A RealtimeDocument is like a normal Document from Kuzzle except that it is
+ * connected to the realtime engine and it's content will change with changes
+ * occuring on the database.
+ *
+ * Realtime documents are resources that should be disposed either with the
+ * stop() or the dispose() method otherwise subscriptions will never be
+ * terminated, documents will be keep into memory and you will end with a
+ * memory leak.
+ */
+export class Observer {
+  /**
+   * Map used to keep track of the observed documents ids by collections.
+   */
+  private documentsBycollections = new Map<CollectionUrn, ObservedDocuments>();
+
+  /**
+   * Map containing the list of realtime documents managed by this observer.
+   *
+   * This map is used to update realtime documents content when notifications
+   * are received.
+   */
+  private documents = new Map<DocumentUrn, RealtimeDocument>();
+  private sdk: Kuzzle;
+
+  constructor (sdk: Kuzzle) {
+    Reflect.defineProperty(this, 'sdk', {
+      value: sdk
+    });
+  }
+
+  /**
+   * Stop observing a list of documents or all the realtime documents
+   * of a collection.
+   *
+   * @param index Index name
+   * @param collection Collection name
+   * @param documents Array of documents (optional)
+   */
+  stop (index: string, collection: string, documents?: Array<{ _id: string }>): Promise<void> {
+    const observedDocuments = this.documentsBycollections.get(collectionUrn(index, collection));
+
+    if (! documents) {
+      for (const id of observedDocuments.ids()) {
+        this.documents.delete(documentUrn(index, collection, id));
+      }
+
+      this.documentsBycollections.delete(collectionUrn(index, collection))
+
+      return this.sdk.realtime.unsubscribe(observedDocuments.roomId);
+    }
+
+    for (const document of documents) {
+      const urn = documentUrn(index, collection, document._id);
+      const rtDocument = this.documents.get(urn);
+
+      if (! rtDocument) {
+        continue;
+      }
+
+      observedDocuments.delete(document._id);
+    }
+
+    return this.resubscribe(index, collection);
+  }
+
+  /**
+   * Unsubscribe from every collections and clear all the realtime documents.
+   */
+  dispose () {
```

Comment:
since there is a stop method, i would call it `stopAll ()` instead wdyt ?

Reply:
Actually I was thinking about having only one method that can take index & collection, index & collection & documents or nothing to stop subscriptions.  
I will update and you can tell me